### PR TITLE
Components for demo

### PR DESCRIPTION
**Pull request by ndido98**
This PR adds the components required for the demo.

### CODE REVIEW

**Review by nicolasfara — APPROVED** [review]

File: demo/src/main/scala/dev/atedeg/ecscalademo/Color.scala [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package dev.atedeg.ecscalademo
+
+case class Color(r: Int, g: Int, b: Int) {
```
Comment (vitlinda) [inline demo/src/main/scala/dev/atedeg/ecscalademo/Color.scala:3]: Should we have an entire module just for this case class ?

Comment (vitlinda) [inline demo/src/main/scala/dev/atedeg/ecscalademo/Color.scala:3]: Should we have an entire module just for this case class ?

**Review by vitlinda — CHANGES_REQUESTED** [review]

File: demo/src/main/scala/dev/atedeg/ecscalademo/Color.scala (hunk 1 above)
Comment (giacomocavalieri) [reply demo/src/main/scala/dev/atedeg/ecscalademo/Color.scala:3]: I agree with Linda, we could move it in the `Components.scala` file with all the other components

File: demo/src/main/scala/dev/atedeg/ecscalademo/Math.scala [hunk 2]
```diff
@@ -24,4 +24,4 @@ extension (scalar: Double) {
 
 extension [T](number: T)(using ord: Ordering[T]) {
   def clamped(min: T, max: T) = if ord.gt(number, max) then max else if ord.lt(number, min) then min else number
-}
\ No newline at end of file
+}
```
Comment (giacomocavalieri) [inline demo/src/main/scala/dev/atedeg/ecscalademo/Math.scala]: Why using the `number, min, max` if `T` can be anything that has an `Ordering`? I think a better naming convention could be `element, upperBound, lowerBound` or something like this.

File: demo/src/test/scala/dev/atedeg/ecscalademo/ColorTest.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      an[IllegalArgumentException] should be thrownBy Color(0, 256, 0)
+    }
+    "have blue capped between 0 and 255" in {
+      noException should be thrownBy Color(0, 0, 0)
+      noException should be thrownBy Color(0, 0, 128)
+      noException should be thrownBy Color(0, 0, 255)
+      an[IllegalArgumentException] should be thrownBy Color(0, 0, -1)
+      an[IllegalArgumentException] should be thrownBy Color(0, 0, 256)
+    }
+  }
```
Comment (giacomocavalieri) [inline demo/src/test/scala/dev/atedeg/ecscalademo/ColorTest.scala:30]: Maybe this code could be refactored a little bit; the only thing changing is the color being tested

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

**Review by vitlinda — APPROVED** [review]

**Pull request merged**